src/Inkluzitron/Modules/Vote/VoteService.cs

```diff
@@ -0,0 +1,331 @@
+                return;
+
+            await ReplySemaphore.WaitAsync();
+
+            try
+            {
+                existingVoteReply = await LocateVoteReplyTo(voteCommandMessage.Channel, voteCommandMessage.Id);
+                if (existingVoteReply is not null)
+                {
+                    await existingVoteReply.ModifyAsync(p => p.Content = voteReplyText);
+                    return;
+                }
+
+                var voteCommandReference = new MessageReference(
+                    voteCommandMessage.Id,
+                    voteCommandMessage.Channel.Id,
+                    guildId
+                );
+
+                var newVoteReply = await voteCommandMessage.Channel.SendMessageAsync(
+                    text: voteReplyText,
+                    allowedMentions: AllowedMentions.None,
+                    messageReference: voteCommandReference
+                );
+
+                using var dbContext = DbFactory.Create();
+                dbContext.VoteReplyRecords.Add(new VoteReplyRecord
+                {
+                    GuildId = guildId,
+                    ChannelId = voteCommandReference.ChannelId,
+                    MessageId = voteCommandMessage.Id,
+                    ReplyId = newVoteReply.Id
+                });
+
+                await dbContext.SaveChangesAsync();
+            }
+            finally
+            {
+                ReplySemaphore.Release();
+            }
+        }
+
+        public async Task ProcessVoteCommandAsync(IUserMessage voteCommandMessage, string voteDefinitionText)
+        {
+            if (voteCommandMessage.Channel is not IGuildChannel guildChannel)
+                return;
+
+            var parse = await Parser.TryParse(new CommandContext(Client, voteCommandMessage), voteDefinitionText);
+
+            if (!parse.Success)
+            {
+                await UpdateVoteReplyAsync(voteCommandMessage, parse.ProblemDescription);
+                return;
+            }
+
+            var allEmotes = new HashSet<IEmote>(parse.Definition.Options.Keys);
+            allEmotes.UnionWith(voteCommandMessage.Reactions.Keys);
+            var failedEmotes = new List<IEmote>();
+            var summary = ComposeSummary(voteCommandMessage, parse.Definition);
+
+            foreach (var emote in allEmotes)
+            {
+                var shouldBePresent = parse.Definition.Options.ContainsKey(emote);
+                var isPresent = voteCommandMessage.Reactions.TryGetValue(emote, out var reaction);
+
+                if (!shouldBePresent && isPresent)
+                {
+                    await voteCommandMessage.RemoveAllReactionsForEmoteAsync(emote);
+                }
+                else if (shouldBePresent && !isPresent)
+                {
+                    try
+                    {
+                        await voteCommandMessage.AddReactionAsync(emote);
+                    }
+                    catch
```

Comment:
Vážně chceš chytat všechny exceptions? :hmmnote: Osobně bych byl za vyjmenování těch exceptions.

Reply:
IMHO je konkrétní typ výjimky implementační detail Discord.NET knihovny, pro účely VoteService je důležité vědět, zda k nějaké výjimce došlo.
Přidal jsem tam catch na HttpException, ale imho to stejně nic neřeší. Nedozvím se, jestli to byl error 503, 401, nebo whatever.

Reply:
Jaktože ne? Property na HttpStatusCode tam je. https://github.com/discord-net/Discord.Net/blob/dev/src/Discord.Net.Core/Net/HttpException.cs#L19